Tests/SphericalExtractionUniformTest/SphericalExtractionUniformTest.cpp

```diff
@@ -0,0 +1,268 @@
+    spherical_extraction_lo.write_extraction("ExtractionOutLo_");
+
+    // high resolution spherical extraction
+    spherical_extraction_params_t extraction_params_hi = extraction_params_lo;
+    // we are only checking the converence in theta integration
+    // extraction_params_hi.num_points_phi *= 2;
+    extraction_params_hi.num_points_theta *= 2;
+    SphericalExtractionUniform spherical_extraction_hi(
+        extraction_params_hi, sim_params.coarsest_dx * sim_params.dt_multiplier,
+        0.0, true, 0.0);
+    spherical_extraction_hi.add_var(c_phi_Re);
+    spherical_extraction_hi.add_var(c_phi_Im);
+    spherical_extraction_hi.extract(&interpolator);
+    spherical_extraction_hi.write_extraction("ExtractionOutHi_");
+
+    // real part is the zeroth componenent and imaginary part is first component
+    auto extracted_harmonic = [](std::vector<double> &data, double, double,
+                                 double) {
+        return std::make_pair(data[0], data[1]);
+    };
+
+    // add the spherical harmonic mode integrands for each resolution
+    // Always use trapezium rule in phi as this is periodic
+    bool broadcast_integral = true;
+    spherical_extraction_lo.add_mode_integrand(
+        sim_params.es, sim_params.el, sim_params.em, extracted_harmonic,
+        integral_lo, method, IntegrationMethod::trapezium, broadcast_integral);
+    spherical_extraction_hi.add_mode_integrand(
+        sim_params.es, sim_params.el, sim_params.em, extracted_harmonic,
+        integral_hi, method, IntegrationMethod::trapezium, broadcast_integral);
+
+    // do the surface integration
+    spherical_extraction_lo.integrate();
+    spherical_extraction_hi.integrate();
+}
+
+int runSphericalExtractionUniformTest(int argc, char *argv[])
+{
+    // Load the parameter file and construct the SimulationParameter class
+    // To add more parameters edit the SimulationParameters file.
+    std::string in_string = argv[argc - 1];
+    pout() << in_string << std::endl;
+    char const *in_file = argv[argc - 1];
+    GRParmParse pp(0, argv + argc, NULL, in_file);
+    SimulationParameters sim_params(pp);
+
+    GRAMR gr_amr;
+    DefaultLevelFactory<SphericalExtractionUniformTestLevel>
+        surface_extraction_test_level_fact(gr_amr, sim_params);
+    // the initial data for the two variables is the spherical harmonic
+    // specified by params
+    setupAMRObject(gr_amr, surface_extraction_test_level_fact);
+
+    AMRInterpolator<Lagrange<4>> interpolator(
+        gr_amr, sim_params.origin, sim_params.dx, sim_params.boundary_params);
+
+    std::pair<std::vector<double>, std::vector<double>> integral_lo_midpoint,
+        integral_hi_midpoint;
+    spherical_extraction_params_t extraction_params_lo_midpoint =
+        sim_params.extraction_params_lo;
+    // must be multiple of 2
+    extraction_params_lo_midpoint.num_points_theta =
+        std::ceil(extraction_params_lo_midpoint.num_points_theta / 2) * 2;
```

Comment:
It should probably print out that it changes the value from the parameter (if it does).